### PR TITLE
drivers: lora: sx127x: fix missing reset gpio instantiation

### PR DESCRIPTION
**Pull request by gadrou**
…e tree.

It seems to me that the lora functions cannot be executed correctly without initializing the reset pin for the SX1272 and SX1276.
Tested on a dedicated board integrating a SX1272.

Signed-off-by: Adrien Rouault <adrien.rouault@nemeus.fr>

### CODE REVIEW

File: drivers/lora/sx127x.c [hunk 1]
```diff
@@ -176,6 +178,9 @@ struct sx127x_config {
 
 static const struct sx127x_config dev_config = {
 	.bus = SPI_DT_SPEC_INST_GET(0, SPI_WORD_SET(8) | SPI_TRANSFER_MSB, 0),
+#if DT_INST_NODE_HAS_PROP(0, reset_gpios)
```
Comment (JordanYates) [inline drivers/lora/sx127x.c:181]: Same here, the property needs to exist so the `#if` guard is not necessary

**Review by JordanYates — CHANGES_REQUESTED** [review]
Please have a look at the format of other commit messages and follow their formatting.
The current commit message also does not match what is actually done in the commit.

**Comment by JordanYates** [comment]
Please have a read through https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-guidelines
Each commit should contain one logical change, and not fix up problems from other commits in the PR.
In general, when addressing feedback you should be modifying existing commits, not creating new ones.

Specifically for this PR, you should squash all 3 commits down to 1, update the commit message, and force push to your branch.

**Comment by JordanYates** [comment]
Almost there, you just need to rebase your branch on top of the latest `main` to get rid of the out-of-date warning

**Review by JordanYates — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Comment by carlescufi** [comment]
@Mani-Sadhasivam could you please take a look?

**Review by Mani-Sadhasivam — APPROVED** [review]

**Comment by Mani-Sadhasivam** [comment]
@carlescufi you can merge now!

**Comment by carlescufi** [comment]
> @carlescufi you can merge now!

Thanks!

**Pull request merged**